course/include/Vector3.h

```diff
@@ -0,0 +1,49 @@
+#ifndef VECTOR3_H
+#define VECTOR3_H
+#include <iostream>
+#include "math.h"
+
+namespace ekumen
+{
+
+class Vector3
+{
+public:
+    explicit Vector3(double x, double y, double z) : x_(x), y_(y), z_(z) {}
+    explicit Vector3() : x_(0.0), y_(0.0), z_(0.0) {}
+    double x() const { return x_; }
+    double y() const { return y_; }
+    double z() const { return z_; }
+        
+    double norm() const;
+    double dot(const Vector3 &q) const;
+    Vector3 cross(const Vector3 &q) const;
+    //operator overload
+    const double& operator[](int index) const;
+    double& operator[](int index);    
+    
+
+    friend Vector3 operator+(const Vector3 &p, const Vector3 &q);
+    friend Vector3 operator-(const Vector3 &p, const Vector3 &q);
```

Comment:
For addition, subtraction and product you should also implement their with-assignment versions (+=, for instance). Add them and then reimplement their without-assignment version to use them instead of implementing the operation twice.